cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -111,32 +107,36 @@ func NewCloudControllerManagerOptions() (*CloudControllerManagerOptions, error)
 }
 
 // NewDefaultComponentConfig returns cloud-controller manager configuration object.
-func NewDefaultComponentConfig(insecurePort int32) (componentconfig.CloudControllerManagerConfiguration, error) {
+func NewDefaultComponentConfig(insecurePort int32) (*componentconfig.CloudControllerManagerConfiguration, error) {
+	// TODO: This code will be fixed up/improved when the ccm API types are moved to their own, real API group out of
+	// pkg/apis/componentconfig to cmd/cloud-controller-manager/app/apis/
 	scheme := runtime.NewScheme()
 	if err := componentconfigv1alpha1.AddToScheme(scheme); err != nil {
-		return componentconfig.CloudControllerManagerConfiguration{}, err
+		return nil, err
 	}
 	if err := componentconfig.AddToScheme(scheme); err != nil {
-		return componentconfig.CloudControllerManagerConfiguration{}, err
+		return nil, err
 	}
+	scheme.AddKnownTypes(componentconfigv1alpha1.SchemeGroupVersion, &componentconfigv1alpha1.CloudControllerManagerConfiguration{})
+	scheme.AddKnownTypes(componentconfig.SchemeGroupVersion, &componentconfig.CloudControllerManagerConfiguration{})
 
-	versioned := componentconfigv1alpha1.CloudControllerManagerConfiguration{}
-	scheme.Default(&versioned)
-
-	internal := componentconfig.CloudControllerManagerConfiguration{}
-	if err := scheme.Convert(&versioned, &internal, nil); err != nil {
+	versioned := &componentconfigv1alpha1.CloudControllerManagerConfiguration{}
+	internal := &componentconfig.CloudControllerManagerConfiguration{}
+	scheme.Default(versioned)
+	if err := scheme.Convert(versioned, internal, nil); err != nil {
 		return internal, err
 	}
-	internal.KubeCloudShared.Port = insecurePort
+	internal.Generic.Port = insecurePort
 	return internal, nil
 }
 
 // Flags returns flags for a specific APIServer by section name
 func (o *CloudControllerManagerOptions) Flags() (fss apiserverflag.NamedFlagSets) {
-	o.CloudProvider.AddFlags(fss.FlagSet("cloud provider"))
-	o.Debugging.AddFlags(fss.FlagSet("debugging"))
-	o.GenericComponent.AddFlags(fss.FlagSet("generic"))
+	// TODO: Fix the --controllers flag for the ccm
```

Comment:
what does this mean?

Reply:
That it will be implemented fully in the future when there is a known list of controllers